moduleframework/mtf_scheduler.py

```diff
@@ -0,0 +1,176 @@
+    #   write: os.environ['MODULE'] 
+
+    # MODULE could be from:
+    #   1. common.get_module_type() ... it reads config.yaml and treceback if it doesn't exist
+    #   2. environment ... MODULE=docker etc
+    #   3. argument ... --module=docker
+    try:
+        args.module=common.get_module_type()
+        # TODO it wrongly writes: 'Using default minimal config ...', change in common.py
+    except moduleframework.mtfexceptions.ModuleFrameworkException:
+        pass
+
+    if os.environ.get('MODULE') is not None:
+        # environment is the highest priority because mtf uses environment (too much)
+        args.module=os.environ['MODULE']
+
+    if args.module in common.get_backend_list():
+        os.environ['MODULE']=args.module
+    else:
+        common.print_info("MODULE={0} ; we support {1} \n === expecting your magic, enjoy! === \n"
+                          "{2}".format(os.environ.get('MODULE'), common.get_backend_list(), parser.epilog))
+        # TODO: what to do here? whats the defaults value for MODULE, do I know it?
+        # exit(1)
+
+    common.print_debug("MODULE={0}".format(os.environ.get('MODULE')))
+
+    # uses additional arguments
+    if args.debug:
+        os.environ['AVOCADO_LOG_DEBUG']='yes'
+        os.environ['DEBUG']='yes'
+    if args.config:
+        os.environ['CONFIG']=args.config
+    if args.url:
+        os.environ['URL']=args.url
+    if args.modulemdurl:
+        os.environ['MODULEMDURL']=args.modulemdurl
+
+    return args, unknown
+
+class AvocadoStart(object):
+    def __init__(self, args, unknown):
+
+        self.module = os.environ['MODULE']
+        self.action = args.action
+
+        # its used for filepath in loadCli:
+        self.tmppath = tempfile.mktemp()
+        site_libs = os.path.dirname(moduleframework.__file__)
+        mtf_tools = "tools"
+
+        # choose between TESTS and ADDITIONAL ENVIRONMENT from options
+        self.tests = ''
+        if args.linter:
+            self.tests = ("{SITE_LIB}/{MTF_TOOLS}/*.py".format(SITE_LIB=site_libs, MTF_TOOLS=mtf_tools))
+        self.additionalAvocadoArg = ''
+        self.returncode=0
+
+        for param in unknown:
+            # take care of this, see tags for safe/unsafe:
+            # http://avocado-framework.readthedocs.io/en/52.0/WritingTests.html#categorizing-tests
+            if os.path.exists(param):
+                # this is list of tests in local file
+                self.tests += " {0} ".format(param)
+            else:
+                # this is additional avocado param
+                self.additionalAvocadoArg += " {0} ".format(param)
+
+        common.print_debug("tests = {0}".format(self.tests))
+        common.print_debug("additionalAvocadoArg = {0}".format(self.additionalAvocadoArg))
+
+    def bash_command(self, cmd):
+        # only to run bash command
+        common.print_debug("cmd: {0}".format(cmd))
+        return process.run(cmd, shell=True, ignore_status=True)
```

Comment:
Why there is `ignore_status=True` as default. in case for example some command isn't found, can cause that it leads to __false negative__ , better is to ensure that all commands pass, and in case you expect that it fails because of some reason you should check return code. so this make sense to ignore this probably just with "avocado run" command what then report exit status to last command like `exit(RC)`

Reply:
the whole tool is designed mainly for avocado run, which can return non zero return code. Such a results is parsed by next tool for logging errors, thus ignore_status=True is needed.

Reply:
yep, it make sense, so it case you can remove this method `bash_command` at all, and use `process.run("avocado what you want", ignore_status=True)` directly in your code, this method is just simple wrapper around `process.run` (btw, you can remove `shell=True` from that) if you don't want to call some bash specific things, (eg. `shell=True` makes sense in case you would like to use command with pipes or use some expansions like `avocado ... *.py ` or `avocado run ... | grep output` what is not your case)